CONTRIBUTING.md

```diff
@@ -70,21 +91,58 @@ Setting Up Your Environment
   itself, the "fast forward" pull will update your local branch without adding
   a merge commit.
 
-7. If you need to update a local branch you are working on, you may "pull" the
-  upstream branch or rebase:
+  (If you have made any local conflicting changes to the `develop` branch itself,
+  then you're doing it wrong.  The use of the `--ff-only` option will detect this
+  for you and refuse to work.  As per point (5) above, all your changes should
+  be happening in a different branch other than `develop`.)
+
+7. If the upstream develop branch changes while you are working on your branch,
+  so you need to update your branch, you may (and probably should) merge the
+  upstream `develop` branch directly into it or rebase:
   ```
+  git checkout whatever-the-branch-i-am-working-on-is-called
   git pull upstream develop
```

Comment:
I added the line because the command `git pull upstream develop` is context dependent.  It pulls into whatever the current branch happens to be.  The fact that we meant for the command to used from the context of "do this when in your local branch you're working on" was easiest to explain by just explicitly showing it.  It's sort of like when shell command examples in instructions often contain a command to `cd` into the directory you're probably already in anyway.  It's there to ensure that the context of where you are meant to execute the command from is crystal clear.